qiita_pet/handlers/api_proxy/studies.py

```diff
@@ -168,3 +171,72 @@ def study_prep_get_req(study_id, user_id):
     return {'status': 'success',
             'message': '',
             'info': prep_info}
+
+
+def study_files_get_req(study_id, prep_template_id, artifact_type):
+    """Returns the uploaded files for the study id categorized by artifact_type
+
+    It retrieves the files uploaded for the given study and tries to do a
+    guess on how those files should be added to the artifact of the given
+    type. Uses information on the prep template to try to do a better guess.
+
+    Parameters
+    ----------
+    study_id : int
+        The study id
+    prep_template_id : int
+        The prep template id
+    artifact_type : str
+        The artifact type
+
+    Returns
+    -------
+    dict of {str: object}
+        A dict of the form {'status': str,
+                            'message': str,
+                            'remaining': list of str,
+                            'file_types': list of (str, bool, list of str),
+                            'num_prefixes': int}
+        where 'status' is a string specifying whether the query is successfull,
+        'message' is a human-readable description of the error (optional),
+        'remaining' is the list of files that could not be categorized,
+        'file_types' is a list of the available filetypes, if it is required
+        or not and the list of categorized files for the given artifact type
+        and 'num_prefixes' is the number of different run prefix values in
+        the given prep template.
+    """
+    supp_file_types = supported_filepath_types(artifact_type)
+    selected = []
+    remaining = []
+
+    uploaded = get_files_from_uploads_folders(study_id)
+    pt = PrepTemplate(prep_template_id).to_dataframe()
+
+    if (any(ft.startswith('raw_') for ft, _ in supp_file_types) and
+            'run_prefix' in pt.columns):
+        prep_prefixes = tuple(set(pt['run_prefix']))
```

Comment:
What about `tuple(pt['run_prefixes'].unique())`? Embracing pandas, not sure it's faster or anything, but the intent is clearer.